instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -29,30 +76,41 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
             attributes.merge!(OpenTelemetry::Instrumentation::Redis.attributes)
 
-            unless config[:db_statement] == :omit
+            # Parse commands if we need to derive span attributes from them
+            if config[:db_statement] != :omit
               parsed_commands = parse_commands(commands)
               parsed_commands = OpenTelemetry::Common::Utilities.truncate(parsed_commands, MAX_STATEMENT_LENGTH)
               parsed_commands = OpenTelemetry::Common::Utilities.utf8_encode(parsed_commands, binary: true)
               attributes['db.statement'] = parsed_commands
             end
 
-            span_name = if commands.length == 1
-                          commands[0][0].to_s.upcase
-                        else
-                          'PIPELINED'
-                        end
-
-            tracer.in_span(span_name, attributes: attributes, kind: :client) do |s|
-              super(commands).tap do |reply|
-                if reply.is_a?(::Redis::CommandError)
-                  s.record_exception(reply)
-                  s.status = Trace::Status.error(reply.message)
-                end
-              end
+            if config[:record_value_size]
+              attributes['db.set_value_size_bytes'] = sent_value_size(commands, SET_VALUE_SIZE_COMMANDS)
             end
+
+            attributes
           end
 
-          private
+          def calculate_bytesize(value) # rubocop:disable Metrics/CyclomaticComplexity
+            case value
+            when ::Redis::CommandError
+              # Don't report on these; call it zero.
+              0
+            when String
+              value.b.bytesize
```

Comment:
Could do more research w/r/t if this is actually necessary — it is an extra string allocation. Per https://github.com/open-telemetry/opentelemetry-ruby/pull/1150/files#r847575538, redis is calling `.b` under the hood before sending over the wire.